src/lib/grammar_ast.rs

```diff
@@ -0,0 +1,171 @@
+use std::collections::{HashMap, HashSet};
+use std::fmt;
+
+pub struct GrammarAST {
```

Comment:
Yes, Abstract Syntax Tree. It is a tree of nonterminals and terminals (not a very deep tree, I admit, but still a tree). In essence what we're writing is a mini-compiler for Yacc. So it's doing the standard parsing -> AST -> other representations. The name isn't perfect, but it has some precedence I think.